app/packages/components/src/components/Toast/Toast.tsx

```diff
@@ -2,20 +2,57 @@ import React from "react";
 import { atom, useRecoilState } from "recoil";
 import { Box, Snackbar, SnackbarContent } from "@mui/material";
 
-// Define types for the props
 interface ToastProps {
   message: React.ReactNode;
-  primary: (setOpen: React.Dispatch<React.SetStateAction<boolean>>) => React.ReactNode;
-  secondary: (setOpen: React.Dispatch<React.SetStateAction<boolean>>) => React.ReactNode;
-  duration?: number;         // Optional duration, with a default value
+  primary?: any;
+  secondary?: any;
+  duration?: number;
+  layout?: {
+    vertical?: "top" | "bottom";
+    horizontal?: "left" | "center" | "right";
+    height?: number | string;
+    top?: number | string;
+    bottom?: number | string;
+    backgroundColor?: string;
+    color?: string;
+    fontSize?: string;
+    textAlign?: string;
+  };
 }
 
 const toastStateAtom = atom({
   key: "toastOpenState",
   default: true,
 });
 
-const Toast: React.FC<ToastProps> = ({message, primary, secondary, duration = 5000 }) => {
+const Toast: React.FC<ToastProps> = ({
+  message,
+  primary,
+  secondary,
+  duration = 5000,
+  layout = {},
+}) => {
+  const snackbarStyle = {
+    height: layout?.height || 5,
+    ...(layout?.top && {
+      top: {
+        xs: layout.top,
```

Comment:
Are these values supposed to be the same? Just checking :)

Reply:
Yeah this will conditionally add `top` or `bottom` **only if** they exist coming in from the Python Panel context